### PR TITLE
sql: fix some string functions

### LINKED ISSUE #2534: sql: re-examine string builtin functions
We need to determine which string functions should also be implemented for `bytes` parameters. For example, `trim`, `strpos`, etc. And we need to re-examine how the existing string functions work with positions. Currently, they operate on byte positions, but perhaps they should utilize character positions and leave byte positions for functions that operate on `bytes`.

### PR DESCRIPTION
**Pull request by tamird**
Closes #2534.

I think implementing functions on bytes belongs in #2042.

### CODE REVIEW

File: sql/parser/builtins.go [hunk 1]
```diff
@@ -206,7 +207,12 @@ var builtins = map[string][]builtin{
 
 	// TODO(XisiHuang): support the position(substring in string) syntax.
 	"strpos": {stringBuiltin2(func(s, substring string) (Datum, error) {
-		return DInt(strings.Index(s, substring) + 1), nil
+		index := strings.Index(s, substring)
+		if index < 0 {
+			return DInt(0), nil
+		}
+
+		return DInt(len([]rune(s[:index])) + 1), nil
```
Comment (petermattis) [inline sql/parser/builtins.go]: Might be better to use `utf8.RuneCountInString(s[:index])`.

File: sql/testdata/builtin_function [hunk 2]
```diff
@@ -213,14 +223,14 @@ SELECT initcap('THOMAS')
 Thomas
 
 query T
-SELECT left('abcde', 2)
+SELECT left('💩abcde', 2)
```
Comment (petermattis) [inline sql/testdata/builtin_function]: Interesting choice of multi-byte unicode character.

**Comment by petermattis** [comment]
LGTM

File: sql/parser/builtins.go (hunk 1 above)
Comment (tamird) [inline sql/parser/builtins.go]: Ah, didn't know about that. Done.

File: sql/parser/builtins.go [hunk 3]
```diff
@@ -216,7 +222,7 @@ var builtins = map[string][]builtin{
 			return DString(strings.Trim(s, chars)), nil
 		}, DummyString),
 		stringBuiltin1(func(s string) (Datum, error) {
-			return DString(strings.Trim(s, " ")), nil
+			return DString(strings.TrimSpace(s)), nil
```
Comment (bdarnell) [inline sql/parser/builtins.go]: This is a deviation from postgres, which only trims the space character, not other whitespace. I don't really care which definition we're using but if we're changing behavior it should be deliberate.

Comment (tamird) [inline sql/parser/builtins.go]: I'm OK with this change for now.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**